src/code/z_actor.c

```diff
@@ -457,439 +1254,3573 @@ f32 Actor_XZDistanceToPoint(Actor* actor, Vec3f* point) {
+    } else {
+        Matrix_SetStateRotationAndTranslation(actor->world.pos.x,
+                                              actor->world.pos.y + (actor->shape.yOffset * actor->scale.y),
+                                              actor->world.pos.z, &actor->shape.rot);
+    }
+
+    Matrix_Scale(actor->scale.x, actor->scale.y, actor->scale.z, MTXMODE_APPLY);
+    Actor_SetObjectDependency(globalCtx, actor);
+
+    gSPSegment(POLY_OPA_DISP++, 0x06, globalCtx->objectCtx.status[actor->objBankIndex].segment);
+    gSPSegment(POLY_XLU_DISP++, 0x06, globalCtx->objectCtx.status[actor->objBankIndex].segment);
+
+    if (actor->colorFilterTimer != 0) {
+        s32 temp_v0_2 = actor->colorFilterParams & 0xC000;
+        Color_RGBA8 actorDefaultHitColor = { 0, 0, 0, 255 };
+
+        if (temp_v0_2 == 0x8000) {
+            actorDefaultHitColor.r = actorDefaultHitColor.g = actorDefaultHitColor.b =
+                ((actor->colorFilterParams & 0x1F00) >> 5) | 7;
+        } else if (temp_v0_2 == 0x4000) {
+            actorDefaultHitColor.r = ((actor->colorFilterParams & 0x1F00) >> 5) | 7;
+        } else if (temp_v0_2 == 0xC000) {
+            actorDefaultHitColor.b = actorDefaultHitColor.g = actorDefaultHitColor.r = 0;
+        } else {
+            actorDefaultHitColor.b = ((actor->colorFilterParams & 0x1F00) >> 5) | 7;
+        }
+
+        if (actor->colorFilterParams & 0x2000) {
+            func_800AE778(globalCtx, &actorDefaultHitColor, actor->colorFilterTimer, actor->colorFilterParams & 0xFF);
+        } else {
+            func_800AE434(globalCtx, &actorDefaultHitColor, actor->colorFilterTimer, actor->colorFilterParams & 0xFF);
+        }
+    }
+
+    actor->draw(actor, globalCtx);
+
+    if (actor->colorFilterTimer != 0) {
+        if (actor->colorFilterParams & 0x2000) {
+            func_800AE8EC(globalCtx);
+        } else {
+            func_800AE5A0(globalCtx);
+        }
+    }
+
+    if (actor->shape.shadowDraw != NULL) {
+        actor->shape.shadowDraw(actor, light, globalCtx);
+    }
+    actor->isDrawn = true;
+
+    CLOSE_DISPS(globalCtx->state.gfxCtx);
+}
+
+void func_800B9D1C(Actor* actor) {
+    s32 sfx = actor->sfx;
+
+    if (sfx != 0) {
+        if (actor->unk39 & 2) {
+            Audio_PlaySoundGeneral(sfx, &actor->projectedPos, 4, &D_801DB4B0, &D_801DB4B0, &D_801DB4B8);
+        } else if (actor->unk39 & 4) {
+            play_sound(sfx);
+        } else if (actor->unk39 & 8) {
+            func_8019F128(sfx);
+        } else if (actor->unk39 & 0x10) {
+            func_801A0810(&D_801DB4A4, NA_SE_SY_TIMER - SFX_FLAG, (sfx - 1));
+        } else if (actor->unk39 & 1) {
+            func_8019F1C0(&actor->projectedPos, sfx);
+        }
+    }
+
+    if (sfx) {}
+
+    if (actor->unk39 & 0x40) {
+        func_801A1FB4(3, &actor->projectedPos, 0x27, 1500.0f);
+    }
+
+    if (actor->unk39 & 0x20) {
+        func_801A1FB4(0, &actor->projectedPos, 0x71, 900.0f);
+    }
```

Comment:
```suggestion
    if (actor->unk39 & 0x40) {
        func_801A1FB4(3, &actor->projectedPos, NA_BGM_MUSIC_BOX_HOUSE, 1500.0f);
    }

    if (actor->unk39 & 0x20) {
        func_801A1FB4(0, &actor->projectedPos, NA_BGM_KAMARO_DANCE, 900.0f);
    }
```
While these aren't in the repo yet, these are actually bgm songs. Meaning `actor->unk39` isn't just for `sfx`s, they're for `bgm` as well. Perhaps you could call  `actor->unk39` -> `actor->audioFlags` as an idea. You should update the descrition in `z64actor.h` as well to include `bgm`